### ISSUE TITLE
Invalid Locale Code Found

### ISSUE THREAD

**Issue by AramRafeq** [ant-design/ant-design#25778]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-cyn8w?file=/index.js)

### Steps to reproduce
1. use configProvider
2. load ku_IQ locale
3. as you see kurmanji dialect is used instead of sorani

### What is expected?
It is expected to be something other than **ku_IQ** beacuse  **ku_IQ** is another alternative to **ckb** **Kurdish(Sorani)** language code what this is a kurmanji dialect with code of **kmr** **Kurdish(Kurmanji)** plus it should omit IQ as kurdish language is spoken in several countries   [check this link for standards](https://iso639-3.sil.org/code/kmr?fbclid=IwAR2Mj7met4TCJih8dgpJwz9lr7wj8pDzfwRH5YmbrIAfFtRErEol3cFgWt0)

### What is actually happening?
the locale called **ku_IQ** but its in kurmanji dialect it should be kurdish-sorani with locale name of **ckb_IQ** kurmanji locale code is **kmr**  [check this link for standards](https://iso639-3.sil.org/code/kmr?fbclid=IwAR2Mj7met4TCJih8dgpJwz9lr7wj8pDzfwRH5YmbrIAfFtRErEol3cFgWt0)

| Environment | Info |
|---|---|
| antd | 4.4.3 |
| React | 16.13.1 |
| System | doesnt matter |
| Browser | doesnt matter |

---
I wanted to integrate kurdish language to locales but i found that there is already a kurdish language and it was named as **ku_IQ** which is not how it should be as kurdish language have 2(Sorani, Kurmanji) main dialects with each one is different even in writing most of the open source projects will accept kurdish sorani language package as **ckb** which stands for "Central Kurdish" people North Iraq speak this dialect but in antd locale content is in  kurdmanji dialect labelled as **ku_IQ** 
[Central Kurdish](http://www.localeplanet.com/icu/ckb/index.html) aslo check [ICU Locale “Central Kurdish (Iraq)” (ckb_IQ)](http://www.localeplanet.com/icu/ckb-IQ/index.html), you can see the diffrence in this page [Kurdish Kurmanji-Sorani Kurdish Dictionary](https://glosbe.com/ku/ckb/ )

**Comment by afc163** [comment]
Maybe you can discuss with @ahalimkara author of https://github.com/ant-design/ant-design/pull/9064, and send us PR if you find solution.

**Comment by ahalimkara** [comment]
Hi @AramRafeq bro, i am more than happy to help with this. I chose ku_IQ because this is how it is used in many packages for Kurmanji dialect. But as you know Kurdish language is not widely supported in the software community and thus we don’t have well defined standards. So maybe i was wrong about ku_IQ. What do you suggest for Kurmanji?

**Comment by AramRafeq** [comment]
@ahalimkara yes unfortunately that is true most of the open source packages use **kmr** for kurmanji dialect and **ckb** for sorani dialect, antd supports Internationalization most of the languages end with country code i suggest we have something like **kmr_KU** and **ckb_KU** the ku part will represent kurdistan  i have created pull request on **rc-picker** package changed it to **kmr_KU** check [this pull request on rc-picker](https://github.com/react-component/picker/pull/110) , plus i have fixed it in antd also the same way check this pull request on antd

**Comment by ahalimkara** [comment]
Hi @AramRafeq 
AFAIK locale format is “languageIdentifier_countryCode” unless i am missing something. So not sure “KU” in “kmr_KU” stands for which country. If it is for Kurdistan Region then it should be IQ as it is still part of Iraq (http://www.localeplanet.com/icu/iso3166.html and https://en.m.wikipedia.org/wiki/Kurdistan_Region). Otherwise i think we need to wait a little bit more until we declare our independency. 
Currently we are using “ku_IQ” for Kurmanji in antd. Ideally it should be “kmr_IQ” but for now i think we can keep using ku_IQ as “ku” here is the top level identifier for all Kurdish dialects and Kurmanji dialect is the most common Kurdish dialect (15M native speakers https://en.m.wikipedia.org/wiki/Kurmanji) and also changing this locale would be a breaking change. 
So to summarize: i would keep using “ku_IQ” for Kurmanji and i will add a new locale “ckb_IQ” for Sorani.

**Comment by AramRafeq** [comment]
@ahalimkara  if we went with **IQ** postfix we could end up with something like **ku_TR** TR stands for turkey, makes a bigger mess than what already exists , i would suggest going with **KU** postfix im aware that this is a breaking change 
## what we can do  to solve this issue/Key Notes
* we can make an alias for **ku_IQ** which already exists named **kmr_KU** so both locales are the same and people can use both untill **ku_IQ** is removed
* after when we do this (when this issue is resolved) i will commence implementing **ckb_KU**
* if we go with **ku** postfix we will eliminate creating and duplication locales based on kurdistan region in each country and + kurdistan Iraq reagoin is not the main   part of Great Kurdistan most people in Iraq speak sorani dialect so it would be preferred to use **KU** as the country code this leads more refined locales and more maintainability
* as mentioned before those are some key standards we must follow to make kurdish locale follow a good standard  [639 Identifier Documentation: kmr](https://iso639-3.sil.org/code/kmr?fbclid=IwAR2Mj7met4TCJih8dgpJwz9lr7wj8pDzfwRH5YmbrIAfFtRErEol3cFgWt0)
* as you mentioned [http://www.localeplanet.com/icu/iso3166.html](http://www.localeplanet.com/icu/iso3166.html) this would create a whole lot of a mess we would end up locales like this **ckb_IR**, **kmr_IR**, **ckb_IQ**, **kmr_IQ**,
![](https://i.ibb.co/bQDXSGj/Screenshot-2020-07-28-at-12-45-30.png)
 never to mention **sdh** and luri dialect take that with the fact turkey **Turkish (Turkey) (tr-TR)** doesn't even approve of existing such a language because of racism.

# Final Notes 
let's not dive to a rabbit hole and involve politics in this in summery i think this is wrong to do something like **ku_IQ** because of issues mentioned above we should go with “languageIdentifier_countryCode” and **countryCode = KU** which stands for kurdistan as for the **languageIdentifier** we would have **ckb, kmr, sdh** to make this backword compatable with exisiting **ku_IQ**  we would leave it as its for now and make it alias for **kmr_KU** in the documentation we will say that **ku_IQ** is an alias for **kmr_KU** and will be removed in the future and thats it.

**Comment by ahalimkara** [comment]
Thanks for the reply, so few things:
1- Lets take it easy, keep calm
2- Having different locale for the same language is not a mess and absolutely fine. See English (en_GB, en_US) Arabic (ar_SA, ar_AE, ar_IQ ...) and much more... This is because same language in different countries may have different pronunciation, grammar etc.
3- I am trying to follow the standards, KU is not a country code _yet_
4- Yes there are many racists and fascists but i don’t care them. Just trying to stick to the standards
5- @afc163 i am happy with any solution, i just wrote my opinions

**Comment by AramRafeq** [comment]
@ahalimkara man i'm calm we want the same thing at the end ❤️
1- we are on the same page :)
2- you got a point here but i still think **ku_IQ** is wrong, at least it should be **kmr_IQ** this would make sense if we did not go with **KU** postfix
3- true
4- "Just trying to stick to the standards" same here 
5- same here

**Comment by ahalimkara** [comment]
Yes I agree on 2, ideally it should be `kmr_IQ`, however for the reasons below as i mentioned above i suggested using `ku_IQ`, but not a big deal, either way works fine IMO.
> Ideally it should be “kmr_IQ” but for now i think we can keep using ku_IQ as “ku” here is the top level identifier for all Kurdish dialects and Kurmanji dialect is the most common Kurdish dialect (15M native speakers https://en.m.wikipedia.org/wiki/Kurmanji) and also changing this locale would be a **breaking change**.

**Comment by afc163** [comment]
How about adding new locale `kmr_IQ` and keep `ku_IQ` for compatibility?

**Comment by ahalimkara** [comment]
Sounds good to me.

**Comment by afc163** [comment]
@AramRafeq What's your thought?

**Comment by AramRafeq** [comment]
@afc163
 **kmr_IQ** i think its acceptable 👍 
After we are done with this i will add **ckb_IQ** for sorani dialect
